File: Proyecto-Prog_Avanzada-railway/src/main/java/co/edu/uniquindio/proyecto/proyectotienda/dto/ProductoGetDTO.java
package co.edu.uniquindio.proyecto.proyectotienda.dto;

import co.edu.uniquindio.proyecto.proyectotienda.modelo.Categoria;
import lombok.*;

import java.util.List;
@AllArgsConstructor
@Getter
@Setter
public class ProductoGetDTO {

    private int codigo;
    private String nombre;
    private String descripcion;
    private int unidades;
    private double precio;
    private List<String> imagenes;
    private List<Categoria> categorias;
}

File: Proyecto-Prog_Avanzada_Frontend-master/src/app/modelo/producto-get-dto.ts
export class ProductoGetDTO {
    codigo: number = 0;
    nombre: string = "";
    descripcion: string = "";
    unidades: number = 0;
    precio: number = 0.0;
    imagenes: string[] = [];
    categorias: string[] = [];

    constructor(codigo: number, nombre: string, descripcion: string, unidades: number, precio: number, imagenes: string[], categorias: string[]) {
        this.codigo = codigo;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.unidades = unidades;
        this.precio = precio;
        this.imagenes = imagenes;
        this.categorias = categorias;
    }
}
